### DISCUSSION TITLE
Create Custom Page

### DISCUSSION THREAD

**Discussion by yasergh** [filamentphp/filament#4492] [General]
Hi,
How can i create a custom page and load data from the model and pass the data to view, I try to create a custom page by command 
php artisan make:filament-page UserDetails --resource=UserResource --type=custom

in-class UserDetails what is the name of the function that allows fetching data from the database and how to create variable can I pass to view 

BR

**Comment by danharrin** [comment]
it's a Livewire component, so you can just add a `public User $record` and then access `$record` in the view... make sure the route in `getPages()` has a `/{record}` parameter

**Reply by yasergh** [reply]
Hi,
Thank you for interested to my question.
`class UserTreeView extends Page
{
    protected static string $resource = UserResource::class;

    protected static string $view = 'filament.resources.member.user-resource.pages.user-tree-view';

    public UserTree $row;

}`
How can I load data from the model  , 
BR

**Reply by yasergh** [reply]
when can i found example or demo page like this

**Reply by danharrin** [reply]
If you're calling it `$row`, make sure your route contains `/{row}`

**Reply by yasergh** [reply]
thank you it is working now if I want to make a filter for the UserTree like showing active only ?
